Tests/Adyen Tests/Analytics/AnalyticsEventDataSourceTests.swift

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright (c) 2024 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import XCTest
+@_spi(AdyenInternal) @testable import Adyen
+
+final class AnalyticsEventDataSourceTests: XCTestCase {
+    
+    var sut: AnalyticsEventDataSource!
+
+    override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+        sut = AnalyticsEventDataSource()
+    }
+
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        sut = nil
+    }
+
+    func testAddingInfo() {
+        XCTAssertEqual(sut.infos.count, 0)
+        
+        sut.add(info: AnalyticsEventInfo(component: "test", type: .rendered))
+        XCTAssertEqual(sut.infos.count, 1)
```

Comment:
Do we need to expose infos logs and errors now since we introduces the wrapper object? Can we make them private?